packages/babel-core/test/assumptions.js

```diff
@@ -42,6 +42,14 @@ describe("assumptions", () => {
     });
   });
 
+  it("cannot be disabled by presets", () => {
+    expect(() =>
+      loadOptions({
+        presets: [() => ({ assumptions: { mutableTemplateObject: false } })],
+      }),
+    ).toThrow();
```

Comment:
Nit: add error message assertion here so it doesn't pass on incorrect error messages.